server/types/oid.go

```diff
@@ -0,0 +1,278 @@
+}
+
+// GetSerializationID implements the DoltgresType interface.
+func (b OidType) GetSerializationID() SerializationID {
+	return SerializationID_Oid
+}
+
+// IsUnbounded implements the DoltgresType interface.
+func (b OidType) IsUnbounded() bool {
+	return false
+}
+
+// MaxSerializedWidth implements the DoltgresType interface.
+func (b OidType) MaxSerializedWidth() types.ExtendedTypeSerializedWidth {
+	return types.ExtendedTypeSerializedWidth_64K
+}
+
+// MaxTextResponseByteLength implements the DoltgresType interface.
+func (b OidType) MaxTextResponseByteLength(ctx *sql.Context) uint32 {
+	return 4
+}
+
+// OID implements the DoltgresType interface.
+func (b OidType) OID() uint32 {
+	return uint32(oid.T_oid)
+}
+
+// Promote implements the DoltgresType interface.
+func (b OidType) Promote() sql.Type {
+	return b
+}
+
+// SerializedCompare implements the DoltgresType interface.
+func (b OidType) SerializedCompare(v1 []byte, v2 []byte) (int, error) {
+	if len(v1) == 0 && len(v2) == 0 {
+		return 0, nil
+	} else if len(v1) > 0 && len(v2) == 0 {
+		return 1, nil
+	} else if len(v1) == 0 && len(v2) > 0 {
+		return -1, nil
+	}
+
+	return bytes.Compare(v1, v2), nil
+}
+
+// SQL implements the DoltgresType interface.
+func (b OidType) SQL(ctx *sql.Context, dest []byte, v any) (sqltypes.Value, error) {
+	if v == nil {
+		return sqltypes.NULL, nil
+	}
+	value, err := b.FormatValue(v)
+	if err != nil {
+		return sqltypes.Value{}, err
+	}
+	return sqltypes.MakeTrusted(sqltypes.Text, types.AppendAndSliceBytes(dest, []byte(value))), nil
+}
+
+// String implements the DoltgresType interface.
+func (b OidType) String() string {
+	return "oid"
+}
+
+// ToArrayType implements the DoltgresType interface.
+func (b OidType) ToArrayType() DoltgresArrayType {
+	return OidArray
+}
+
+// Type implements the DoltgresType interface.
+func (b OidType) Type() query.Type {
+	return sqltypes.Uint32
+}
+
+// ValueType implements the DoltgresType interface.
+func (b OidType) ValueType() reflect.Type {
+	return reflect.TypeOf(uint32(0))
+}
+
+// Zero implements the DoltgresType interface.
+func (b OidType) Zero() any {
+	return uint32(0)
+}
+
+// SerializeType implements the DoltgresType interface.
+func (b OidType) SerializeType() ([]byte, error) {
+	return SerializationID_Oid.ToByteSlice(0), nil
+}
+
+// deserializeType implements the DoltgresType interface.
+func (b OidType) deserializeType(version uint16, metadata []byte) (DoltgresType, error) {
+	switch version {
+	case 0:
+		return Oid, nil
+	default:
+		return nil, fmt.Errorf("version %d is not yet supported for %s", version, b.String())
+	}
+}
+
+// SerializeValue implements the DoltgresType interface.
+func (b OidType) SerializeValue(val any) ([]byte, error) {
+	if val == nil {
+		return nil, nil
+	}
+	converted, _, err := b.Convert(val)
+	if err != nil {
+		return nil, err
+	}
+	retVal := make([]byte, 4)
+	binary.BigEndian.PutUint32(retVal, uint32(converted.(uint32))+(1<<31))
```

Comment:
We add the shifted value for integers as it allows us to do a quick byte-comparison in `SerializedCompare`, since the sign bit is the upper-most bit which breaks the natural bit ordering. That doesn't apply to unsigned integers.